src/osd/OSD.cc

```diff
@@ -1503,7 +1503,6 @@ bool OSDService::_get_map_bl(epoch_t e, bufferlist& bl)
 
 bool OSDService::get_inc_map_bl(epoch_t e, bufferlist& bl)
 {
-  std::lock_guard l(map_cache_lock);
```

Comment:
if we remove this lock, we need to check `contents` in `SimpleLRU::_add()` again before adding the new entry to to `lru` and `contents`. we could introduce the semantics of `map::try_emplace()` to load/construct the `bl` on demand to avoid the extra lock (`OSDService::map_cache_lock`) over `SimpleLRU::lock`. but this is not implemented yet in existing implementation or in this patchset.

Reply:
> BTW why add lock in SharedLRU & SimpleLRU?

we could remove the lock in them as long as the atomicity is ensured. you could check all the users in the source tree to see if we protect all of them with locks like `map_cache_lock`. my answer is "not all of them".

Reply:
Thanks very much. I'll check all usage and give a better method.